server/app/services/program/ProgramService.java

```diff
@@ -223,27 +223,26 @@ public ImmutableList<ProgramDefinition> getAllProgramDefinitionVersions(long pro
   }
 
   private CompletionStage<ProgramDefinition> syncProgramAssociations(ProgramModel program) {
-    if (isActiveOrDraftProgram(program)) {
+    VersionModel activeVersion = versionRepository.getActiveVersion();
+    VersionModel maxVersionForProgram =
+        programRepository.getVersionsForProgram(program).stream()
+            .max(Comparator.comparingLong(p -> p.id))
+            .orElseThrow();
+    // If the max version is greater than the active version, it is a draft
+    if (maxVersionForProgram.id > activeVersion.id) {
```

Comment:
**praise:** This is a great example of a "why" comment. I love finding these when I read unfamiliar code.